client/packages/system/src/Vaccination/Components/VaccinationModal.tsx

```diff
@@ -256,33 +229,35 @@ const VaccinationForm = ({
         sx={{ margin: '0 auto' }}
         value={draft.given ?? null}
         onChange={event =>
-          updateDraft({ given: event.target.value === 'true' })
+          updateDraft({
+            given: event.target.value === 'true',
+            // Ensure current facility is selected when changing given status
+            facilityId: isFreeTextFacility ? OTHER_FACILITY : store?.nameId,
```

Comment:
Was a bit too easy to change to given, while leaving the old facility selected. I think this feels like the right behaviour?

Reply:
Nice

Reply:
I think that would still be okay, if you are making edits, we should now associate the record with your facility? There's probably cases where it will end up "wrong" either way. If we are continuing with more vaccination features, I would pretty strongly push that we move towards them being "insert only". So we keep original record from first facility, from first encounter, and create a new one with whatever happens in current encounter. Would be better for tracing and reporting...